pyEpiabm/pyEpiabm/tests/test_unit/test_routine/test_simulation.py

```diff
@@ -232,18 +348,109 @@ def test_s_write_to_file(self, mock_mkdir):
                                   self.sweeps, self.sim_params,
                                   self.spatial_file_params)
             data = {s: 0 for s in list(pe.property.InfectionStatus)}
-            data["age_group"] = len(pe.Parameters.instance().age_proportions)
-            data["time"] = time
-            cell = self.test_population.cells[0]
-            data["cell"] = cell.id
-            data["location_x"] = cell.location[0]
-            data["location_y"] = cell.location[0]
+            for cell in self.test_population.cells:
+                for age_i in \
+                        range(0,
+                              len(pe.Parameters.instance().age_proportions)):
+                    for inf_status in data:
+                        data_per_inf_status = \
+                                cell.compartment_counter.retrieve()[inf_status]
+                        data[inf_status] += data_per_inf_status[age_i]
+                data["age_group"] = \
+                    len(pe.Parameters.instance().age_proportions)
+                data["time"] = time
+                cell = self.test_population.cells[0]
+                data["cell"] = cell.id
+                data["location_x"] = cell.location[0]
+                data["location_y"] = cell.location[0]
 
             with patch.object(spatial_sim.writer, 'write') as mock:
                 spatial_sim.write_to_file(time)
                 mock.assert_called_with(data)
         mock_mkdir.assert_called_with(os.path.join(os.getcwd(),
-                                      self.file_params["output_dir"]))
+                                                   self.file_params[
+                                                       "output_dir"]))
+
+    @patch('os.makedirs')
+    def test_write_to_ih_file(self, mock_mkdir):
```

Comment:
This is a very long test - can it be split into three separate tests or would that require code duplication? Traditionally a unit test only tests one thing - we take some liberties with that in this code base, but even so this is rather large

Reply:
I will have a look at this